frontend/app/src/ui/mod.rs

```diff
@@ -257,17 +266,28 @@ impl UI {
                     self.physics_time += dt;
                     self.update_snapshot(true);
                 }
-            } else if self.single_steps > 0 {
+            } else if self.steps_forward > 0 {
                 self.physics_time += dt;
                 self.update_snapshot(false);
+            } else if self.steps_backward > 0 {
+                // Avoiding overflow
+                self.physics_time = if self.physics_time < dt {
+                    Duration::from_secs(0)
+                } else {
+                    self.physics_time - dt
+                };
+
+                self.update_snapshot(false);
+                self.steps_backward -= 1;
             } else if self.slowmo {
                 self.physics_time += dt / 10;
                 self.update_snapshot(true);
             } else {
+                // TODO: When do we hit this branch? When it's the first snapshot?
                 self.update_snapshot(true);
             }
-            if self.single_steps > 0 {
-                self.single_steps -= 1;
+            if self.steps_forward > 0 {
+                self.steps_forward -= 1;
             }
```

Comment:
I wasn't completely sure why steps forward was decremented down here rather than up above with the rest of the step forward logic. Should I decrement `steps_backward` down here as well?

I'm also curious under what circumstances `steps_forward` is greater than 1?